app/models/vacation.rb

```diff
@@ -2,13 +2,53 @@ class Vacation < ApplicationRecord
   belongs_to :employee
 
   validates :started_on, presence: true
-  validates :finished_on, presence: true
-  validates :employee_id, presence: true, uniqueness: true
-  validate :start_and_finish_dates
-end
+  validates :finished_on, presence: true, comparison: { greater_than: :started_on }
+  validates :employee_id, presence: true
+  validate :remaining_vacation_days
+  validate :not_past_date
+  validate :vacation_overlap
+  validate :position_active
+  validate :can_take_vacation
+
+  def remaining_vacation_days
+    remaining_days = employee.vacation_days - employee.used_vacation_days
+    return if remaining_days > 0
+    errors.add(:vacation_days, "Dont have enough days")
+  end
+
+  def not_past_date
+    return unless started_on && finished_on
+    return if started_on > DateTime.now
+    errors.add(:vacation_days, "You cannot take vacation in a past date")
+  end
 
-def start_and_finish_dates
-  if  started_on && finished_on && started_on > finished_on
-    errors.add(:finished_on, "Vacation cannot end before it begins")
+  def vacation_overlap
+    employee.vacations.each do |vacation|
```

Comment:
Again, you are fetching all vacations of this employee. But you should not. Much easier for db will be the query that just checks for existing.
`employee.vacations.exists?(...)`